### PR TITLE
[Horizon] Tooltip

### PR DESCRIPTION
**Pull request by reabbotted**
The approach was to make calling it nearly identical to .popover, but limit the styling options to the enumerations .primary and .secondary to match the style guide.

Screenshot 2024-12-12 at 3 47 50 PM
Screenshot 2024-12-12 at 3 47 53 PM

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



<details><summary>:x: XCTest failed: CoreTests/AssignmentCellViewModelTests/testSubmissionStatusAndIconAndColor</summary>


    XCTAssertEqual failed: ("#6a7883") is not equal to ("#697783")
    XCTAssertEqual failed: ("#03893d") is not equal to ("#03893c")
</details>
<details><summary>:x: XCTest failed: CoreTests/CourseSyncCleanupInteractorTests/testApplicationOfflineDirectoryForUserDeleted</summary>


    XCTAssertFalse failed
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6dec10cf8cff081f49ae4eedf0905078209c784d
</p>

**Comment by Ahmed-Naguib93** [comment]
I have checked that the primary and secondary buttons are working fine, but the rest of the buttons are not clickable.

**Review by szabinst — CHANGES_REQUESTED** [review]
Like the view modifier approach! 

See my findings below.

File: packages/HorizonUI/Sources/HorizonUI/Sources/Components/Tooltip/HorizonUI.Tooltip.swift [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            case .secondary:
+                return Color.huiColors.surface.cardPrimary
+            }
+        }
+    }
+}
+
+extension HorizonUI {
+    enum Tooltip {
+        struct Tooltip<TooltipContent: View>: ViewModifier {
```
[the remark this answers is not in the document]
Comment (reabbotted) [reply packages/HorizonUI/Sources/HorizonUI/Sources/Components/Tooltip/HorizonUI.Tooltip.swift:66]: I ended up removing the enum Tootip declaration. It's now just HorizonUI.Tooltip as it should have been.

**Review by szabinst — DISMISSED** [review]

**Comment by reabbotted** [comment]
There is a bug in iOS before 18.1 that doesn't honor the edge direction.

Screenshot 2024-12-13 at 1 12 24 PM

**Comment by inst-danger** [comment]
<details><summary>Horizon Build QR Code:</summary>

</details>

**Review by szabinst — APPROVED** [review]

**Pull request merged**